pkg/deploy/kubernetes_image_puller.go

```diff
@@ -379,6 +385,88 @@ func UpdateImagePullerSpecIfEmpty(ctx *DeployContext) (orgv1.CheClusterSpecImage
 	return ctx.CheCluster.Spec.ImagePuller, nil
 }
 
+func AddRelatedImages(ctx *DeployContext) {
+	relatedImages := GetRelatedImagesEnvVars()
+
+	if len(relatedImages) == 0 {
+		return
+	}
+
+	if len(ctx.CheCluster.Spec.ImagePuller.Spec.Images) > 0 &&
+		!strings.HasSuffix(ctx.CheCluster.Spec.ImagePuller.Spec.Images, ";") {
+		ctx.CheCluster.Spec.ImagePuller.Spec.Images += ";"
+	}
+
+	for _, envVar := range relatedImages {
+		ctx.CheCluster.Spec.ImagePuller.Spec.Images += envVar.Name + "=" + envVar.Value + ";"
+	}
+	ctx.ClusterAPI.Client.Update(context.TODO(), ctx.CheCluster, &client.UpdateOptions{})
+}
+
+func GetRelatedImagesEnvVars() []v1.EnvVar {
+	relatedImagePatterns := [...]string{
+		"^RELATED_IMAGE_.*_plugin_java8$",
+		"^RELATED_IMAGE_.*_plugin_java11$",
+		"^RELATED_IMAGE_.*_plugin_kubernetes$",
+		"^RELATED_IMAGE_.*_plugin_openshift$",
+		"^RELATED_IMAGE_.*_plugin_broker.*",
+		"^RELATED_IMAGE_.*_theia.*",
+		"^RELATED_IMAGE_.*_stacks_cpp$",
+		"^RELATED_IMAGE_.*_stacks_dotnet$",
+		"^RELATED_IMAGE_.*_stacks_golang$",
+		"^RELATED_IMAGE_.*_stacks_php$",
+		"^RELATED_IMAGE_.*_cpp_.*_devfile_registry_image.*",
+		"^RELATED_IMAGE_.*_dotnet_.*_devfile_registry_image.*",
+		"^RELATED_IMAGE_.*_golang_.*_devfile_registry_image.*",
+		"^RELATED_IMAGE_.*_php_.*_devfile_registry_image.*",
+		"^RELATED_IMAGE_.*_java.*_maven_devfile_registry_image.*",
+	}
+
+	var err error
+	relatedImages := []v1.EnvVar{}
+	for _, pattern := range relatedImagePatterns {
+		if matches := util.GetEnvByRegExp(pattern); len(matches) > 0 {
+			for _, match := range matches {
+				match.Name = match.Name[len("RELATED_IMAGE_"):]
+				match.Name, err = ConvertToRFC1123(match.Name)
```

Comment:
Could you explain why do we need to convert name to RFC1123 format ?

Reply:
I think Kubernetes/Openshift enforces this for the names, since otherwise, I get this error in the pod's logs:

```
2021/06/18 20:27:32 Failed to create daemonset: DaemonSet.apps "kubernetes-image-puller" is invalid: [spec.template.spec.containers[0].name: Invalid value: "RELATED_IMAGE_keycloak": a lowercase RFC 1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?')
```